watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -112,7 +120,14 @@ def build(include_pretreatment=False):
 
     # define flowsheet inlets and outlets
     m.fs.feed = FeedZO(property_package=m.fs.prop_nf)
-    m.fs.dye_retentate = Product(property_package=m.fs.prop_nf)
+
+    if include_dewatering == True:
+        dewater = m.fs.dewater = Block()
```

Comment:
Can you clarify why you introduced this dewater Block rather than directly add `m.fs.dewater=Separator(...)`. Then for the following arc, you can directly say` if include_dewatering == True:` and define another flowsheet

Reply:
Ironically, I had the same question, but I was just following the convention set by how pretreatment was handled in this flowsheet (which is also a config option that can be toggled on and off). I assume there was a reason why the original flowsheet was set up this way... any insight on this @adam-a-a ?

Reply:
@luohezhiming @adam-a-a Directly using `m.fs.dewaterer = Separator(...)` does not make the MultiUnitModelCostingBlock work - it still runs into the same failure (see https://github.com/watertap-org/watertap/pull/1278#discussion_r1471618168).